### PR TITLE
documentation: Add basic library documentation

### PR DESCRIPTION
**Pull request by vargac**
* Code documentation - mainly the interface
* Model and algorithm description

Signed-off-by: vargac <simon.varga123@gmail.com>

### CODE REVIEW

**Comment by vargac** [comment]
@mrshu Could you have a look at it? Or should I request a review from someone else, or just merge by myself?

**Comment by mrshu** [comment]
Sorry for ingoring this @vargac.

@Adman would you happen to have some time to review this? If not, I'll try my best to take a look but no sooner than next week -- sorry about that.

File: PixyViSy.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ void PixyViSy::processBall()
         return;
     }
     Block& ball = pixy.blocks[ball_i];
-    if (ball.width > ball.height) {
+    if (ball.width > ball.height) { // prevent covered ball (e.g. by robot)
```
Comment (Adman) [inline PixyViSy.cpp]: Put comment before `if` statement, and also between /* and */

File: PixyViSy.cpp [hunk 2]
```diff
@@ -189,6 +187,8 @@ inline int16_t PixyViSy::getPixY(int16_t Y, int16_t Z)
     return v_center - Y * Fyp / Z;
 }
 
+/* Note that atan is not as slow as other goniometric functions */
+
```
Comment (Adman) [inline PixyViSy.cpp]: Remove empty line

File: PixyViSy.h [hunk 3]
```diff
@@ -12,9 +12,12 @@
 class PixyViSy
 {
     public:
+        /* "pixel_Fx"/"Fy" are intristic parameters and "flag" determines
+           what will be processed in "update" function */
```
Comment (Adman) [inline PixyViSy.h]: Please, describe a bit better what can be processed in update function according to `flag` argument. (and maybe what values can we set flag to)

**Review by Adman — CHANGES_REQUESTED** [review]
@vargac Just small nitpicks to fix. Otherwise it looks good to me.

File: PixyViSy.h (hunk 3 above)
Comment (vargac) [reply PixyViSy.h]: What can be processed is described in README, so I don't have to repeat it, do I?

**Review by Adman — APPROVED** [review]

**Pull request merged**